### PR TITLE
feat: use paddle prices and pass country data

### PR DESCRIPTION
**Pull request by asbiin**

### CODE REVIEW

File: app/Http/Controllers/MonicaController.php [hunk 1]
```diff
@@ -14,13 +15,16 @@ public function index(Request $request)
     {
         $plans = Plan::where('product', static::PRODUCT)->get();
```
[the remark this answers is not in the document]
Comment (asbiin) [reply app/Http/Controllers/MonicaController.php:16]: This is local to the class, but let's fix it in another PR...

File: app/Services/ProductPrices.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace App\Services;
+
+use App\Models\User;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\Cache;
+use Laravel\Paddle\ProductPrice;
+
+class ProductPrices
```
[the remark this answers is not in the document]
Comment (asbiin) [reply app/Services/ProductPrices.php:11]: Do you have a proposition name?

File: resources/js/Pages/OfficeLife/Index.vue [hunk 3]
```diff
@@ -165,24 +166,17 @@ export default {
...
     };
   },
 
   methods: {
     checkPrice(plan) {
-      this.form.quantity = plan.quantity;
-
       axios
-        .post(plan.url.price, this.form)
+        .post(route('officelife.price', { plan: plan.id }), { quantity: plan.quantity })
```
Comment (djaiss) [inline resources/js/Pages/OfficeLife/Index.vue:175]: I don't like this, but I won't fight it on this project.
In my opinion we shouldn't call route names in the views directly. This knowledge should come from the backend.

File: app/Services/ProductPrices.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (djaiss) [reply app/Services/ProductPrices.php:12]: `GetProductPrices` would be a better name.

[the remark this answers is not in the document]
Comment (djaiss) [reply app/Services/ProductPrices.php:12]: `GetProductPrices` would be a better name.

[the remark this answers is not in the document]
Comment (djaiss) [reply app/Services/ProductPrices.php:12]: I just realized... it's not a Service, actually. This is... a view helper, or something. But not a service. This is why I was confused. A service does something (update, create, ...). But this only gets data for the view, right?

[the remark this answers is not in the document]
Comment (djaiss) [reply app/Services/ProductPrices.php:12]: I just realized... it's not a Service, actually. This is... a view helper, or something. But not a service. This is why I was confused. A service does something (update, create, ...). But this only gets data for the view, right?

**Review by djaiss — COMMENTED** [review]
This is great overall.

File: resources/js/Pages/OfficeLife/Index.vue (hunk 3 above)
Comment (asbiin) [reply resources/js/Pages/OfficeLife/Index.vue]: There are a lots of `route` calls in vue, I don't understand your point.

**Review by djaiss — APPROVED** [review]

File: resources/js/Pages/OfficeLife/Index.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (asbiin) [reply resources/js/Pages/OfficeLife/Index.vue]: I kindly disagree. We always call route names in the view. On the contrary, calling url directly is really a bad choice.

[the remark this answers is not in the document]
Comment (djaiss) [reply resources/js/Pages/OfficeLife/Index.vue]: I prefer that routes are not built on the front end at all. 

If the URL comes from the backend, it's better as urls are like any other data needed to display and build the view.

What is the benefits from building the url in the view? To me it's just additional logic that the view should not have to worry about.

**Pull request merged**